src/puppetlabs/puppetdb/query/aggregate_event_counts.clj

```diff
@@ -1,50 +1,78 @@
+            [puppetlabs.puppetdb.query.events :as events]
             [puppetlabs.puppetdb.jdbc :as jdbc]
             [puppetlabs.kitchensink.core :as kitchensink]))
 
 (defn- get-aggregate-sql
   "Given the `event-count-sql`, return a SQL string that will aggregate the results."
-  [event-count-sql]
+  [event-count-sql summarize_by]
   {:pre  [(string? event-count-sql)]
    :post [(string? %)]}
-  (format "SELECT SUM(CASE WHEN successes > 0 THEN 1 ELSE 0 END) as successes,
+  (format "(SELECT CAST('%s' AS VARCHAR(20)) as summarize_by,
+           SUM(CASE WHEN successes > 0 THEN 1 ELSE 0 END) as successes,
                   SUM(CASE WHEN failures > 0  THEN 1 ELSE 0 END) as failures,
                   SUM(CASE WHEN noops > 0 THEN 1 ELSE 0 END) as noops,
                   SUM(CASE WHEN skips > 0 THEN 1 ELSE 0 END) as skips,
                   COUNT(*) as total
-           FROM (%s) event_counts" event-count-sql))
+           FROM (%s) event_counts)" summarize_by event-count-sql))
 
-(defn query->sql
-  "Convert an aggregate-event-counts `query` and a value to `summarize_by` into a SQL string.
-  Since all inputs are forwarded to `event-counts/query->sql`, look there for proper documentation."
+(defn assemble-aggregate-sql
+  "Convert an aggregate-event-counts `query` and a value to `summarize_by`
+   into a SQL string."
   [version query [summarize_by query-options]]
   {:pre  [(sequential? query)
           (string? summarize_by)
-          ((some-fn map? nil?) query-options)]
-   :post [(jdbc/valid-jdbc-query? (:results-query %))]}
+          ((some-fn map? nil?) query-options)]}
   (let [query-options (if (nil? query-options) {} query-options)
         [count-sql & params] (:results-query
-                              (event-counts/query->sql version query [summarize_by query-options {}]))
-        aggregate-sql        (get-aggregate-sql count-sql)]
-    {:results-query (apply vector aggregate-sql params)}))
+                              (event-counts/query->sql
+                                :aggregate-event-counts
+                                version
+                                query [summarize_by query-options {}]))]
+    [(get-aggregate-sql count-sql summarize_by) params]))
+
+(defn query->sql
+  "Convert an aggregate-event-counts `query` and a value(s) to `summarize_by`
+   into a SQL string. Since all inputs are forwarded to
+   `event-counts/query->sql`, look there for proper documentation."
+  [version query [summarize_by {:keys [distinct_resources?] :as query-options}]]
+  {:pre  [(sequential? query)
+          ((some-fn map? nil?) query-options)]
+   :post [(jdbc/valid-jdbc-query? (:results-query %))]}
+  (let [summary-vec (str/split summarize_by #",")
+        nsummarized (count summary-vec)
+        aggregate-fn #(assemble-aggregate-sql version query [% query-options])
+        aggregated-sql-and-params (map aggregate-fn summary-vec)
+        common-params (second (first aggregated-sql-and-params))
```

Comment:
nope, currently the params are the same for each -- you can see that from the aggregate-fn definition above. They all have the same query-options and differ only in summarize_by.

If I pull summarize_by into a param I'll adjust this. For the record I think the way these five lines work right now is pretty nasty. Definitely open to suggestions -- the issue is that it's doing something with shared code that none of our other endpoints do.